src/Cli/dotnet/Installer/Windows/MsiPackageCache.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable disable
+
+using System;
+using System.IO;
+using System.IO.Pipes;
+using System.Runtime.Versioning;
+using System.Security.AccessControl;
+using System.Security.Principal;
+using Microsoft.DotNet.Cli.Utils;
+using Newtonsoft.Json;
+
+namespace Microsoft.DotNet.Installer.Windows
+{
+    /// <summary>
+    /// Manages caching workload pack MSI packages.
+    /// </summary>
+    [SupportedOSPlatform("windows")]
+    internal class MsiPackageCache : InstallerBase
```

Comment:
this class looks like interop class too. Could you comment the counter part document?

Reply:
Don't think there's any interop or p/invoke calls here. Could you clarify what you mean by the counter part?